homeassistant/components/melcloud/climate.py

```diff
@@ -169,3 +209,128 @@ def max_temp(self) -> float:
         return convert_temperature(
             DEFAULT_MAX_TEMP, TEMP_CELSIUS, self.temperature_unit
         )
+
+
+ATW_ZONE_HVAC_MODE_LOOKUP = {
+    atw.ZONE_OPERATION_MODE_HEAT: HVAC_MODE_HEAT,
+    atw.ZONE_OPERATION_MODE_COOL: HVAC_MODE_COOL,
+}
+ATW_ZONE_HVAC_MODE_REVERSE_LOOKUP = {v: k for k, v in ATW_ZONE_HVAC_MODE_LOOKUP.items()}
+
+
+class AtwDeviceZoneClimate(MelCloudClimate):
+    """Air-to-Water zone climate device."""
+
+    def __init__(self, device: MelCloudDevice, atw_device: AtwDevice, atw_zone: Zone):
+        """Initialize the climate."""
+        super().__init__(device)
+        self._device = atw_device
+        self._zone = atw_zone
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        return f"{self.api.device.serial}-{self.api.device.mac}-{self._zone.zone_index}"
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this entity."""
+        return f"{self._name} {self._zone.name}"
+
+    @property
+    def state_attributes(self) -> Dict[str, Any]:
+        """Return the optional state attributes with device specific additions."""
+        data = super().state_attributes
+        data[ATTR_STATUS] = ATW_ZONE_HVAC_MODE_LOOKUP.get(
+            self._zone.status, self._zone.status,
+        )
+        return data
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_UNIT_LOOKUP.get(self._device.temp_unit, TEMP_CELSIUS)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        mode = self._zone.operation_mode
+        if not self._device.power or mode is None:
+            return HVAC_MODE_OFF
+        return ATW_ZONE_HVAC_MODE_LOOKUP.get(mode, HVAC_MODE_OFF)
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self._device.set({"power": False})
+            return
+
+        operation_mode = ATW_ZONE_HVAC_MODE_REVERSE_LOOKUP.get(hvac_mode)
+        if operation_mode is None:
+            raise ValueError(f"Invalid hvac_mode [{hvac_mode}]")
+
+        if self._zone.zone_index == 1:
+            props = {PROPERTY_ZONE_1_OPERATION_MODE: operation_mode}
+        else:
+            props = {PROPERTY_ZONE_2_OPERATION_MODE: operation_mode}
+        if self.hvac_mode == HVAC_MODE_OFF:
+            props["power"] = True
+        await self._device.set(props)
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return [self.hvac_mode]
```

Comment:
This makes no sense ? Why would it only ever return it's current one ?

Reply:
Some devices have both heat/cool capabilities and in those rare cases both of the modes would be returned here. The support for those devices is not implemented in the client library because I just haven't found a sample system for the purpose.

The actual available modes are heat and off, but according to the feedback I got it's counter intuitive to control the system level power on/off through the individual zones. Knowing the system layout the off mode on the zone climate looks like it should turn off only the zone.

I wanted the `HVAC_OFF` mode to be displayed on the zones when the system is powered off and this is the cleanest approach I could come up with.